controllers/servers/csi/csi_addons_server/fence_controller_servicer.py

```diff
@@ -0,0 +1,46 @@
+from csi_general import fence_pb2_grpc, fence_pb2
+
+from controllers.array_action.storage_agent import get_agent, detect_array_type
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers import utils
+from controllers.servers.csi.decorators import csi_fence_method
+
+logger = get_stdout_logger()
+
+
+def _is_already_handled(mediator, fence_ownership_group):
+    return mediator.is_fenced(fence_ownership_group)
+
+
+def _fence_cluster_network(mediator, fence_ownership_group, unfence_ownership_group):
+    logger.info("fencing {}".format(fence_ownership_group))
+    mediator.fence(fence_ownership_group, unfence_ownership_group)
+
+
+def handle_fencing(request):
+    fence_ownership_group = request.parameters["fenceToken"]
```

Comment:
what if these parameters are missing, is it handled somewhere?

Reply:
right, done